### ISSUE TITLE
How should users publicly distribute locale models?

### ISSUE THREAD

**Issue by benharsh** [chapel-lang/chapel#8767]
If someone writes their own locale model, how should they allow others to use it? Today it would probably involve copying a file into modules/internal/localeModels/<lm-name>. That's not only brittle and crufty, but users on shared systems may not have permission to perform an operation like that.

Is mason an answer to this question?

**Comment by benharsh** [comment]
@gbtitus , @dmk42 : I welcome any thoughts you have on this topic, feel free to update the issue text.

**Comment by dmk42** [comment]
Maybe we could have a CHPL_USER_LOCALE_MODEL_DIR environment variable, or something shorter.

**Comment by benharsh** [comment]
If we went that route, we could probably just re-use $CHPL_LOCALE_MODEL and point it to a directory. Or maybe the compiler could check $MASON_HOME after looking at $CHPL_HOME/modules/internal/localeModels?

**Comment by bradcray** [comment]
I agree that locale models should ultimately be distributable, like other packages.  I also agree with the implied sentiment that there aren't likely to be a lot of people writing this type of code, particularly in the short-term.

**Comment by gbtitus** [comment]
> ... maybe the compiler could check $MASON_HOME after looking at $CHPL_HOME/modules/internal/localeModels?

This seems like a natural extension and is the option I gave 👍 for in the comment.
